quickwit/quickwit-rest-client/src/rest_client.rs

```diff
@@ -771,9 +801,7 @@ mod test {
         let server_url = Url::parse(&format!("http://127.0.0.1:{port}")).unwrap();
         let qw_client = QuickwitClientBuilder::new(server_url).build();
         let error = qw_client.indexes().list().await.unwrap_err();
-
-        assert!(matches!(error, Error::Client(_)));
-        assert!(error.to_string().contains("tcp connect error"));
+        assert!(matches!(error, Error::Middleware(_)));
```

Comment:
can we assert that we still say something about a tcp connect error? i don't know how this error stringify itself, but if it says nothing about the underlying situation, it's not really useful to the (human) consumer